src/schema/me/recently_viewed_artworks.ts

```diff
@@ -21,3 +26,31 @@ export const RecentlyViewedArtworks = {
     })
   },
 }
+
+export const recordArtworkViewMutation = mutationWithClientMutationId({
+  name: "RecordArtworkView",
+  description: "Records an artwork view.",
+  inputFields: {
+    artwork_id: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  },
+  outputFields: {
+    success: {
```

Comment:
You need to be able to resolve _something_ from the mutation (perhaps just `clientMutationId` would work for that). I spent a while thinking about what clients might want to do here. It seems like returning `artwork`, with the fetched artwork that's being recorded might make sense, so clients can prepend to an already fetched set.

But, it may make sense to actually return `artworks`, which is a proper connection for the recently viewed artworks. It may also make sense to just return the user (what the Gravity endpoint itself returns), and clients can do whatever they want.

So, since you need to resolve _something_, I decided to hold off on adding anything more meaningful here until you need it. Adding `artworks` and resolving that, for instance, would be performing a fetch _after_ the mutation, which is certainly not needed yet.

Reply:
Now that you mention it, what do the various references to `clientMutationId` mean? I see it embedded in names but no actual id value appears.

Reply:
`clientMutationId` is a Relay spec for mutations. It's a field that clients pass in (so it's implicitly an 'input' to every mutation), and the same value passed in must be returned in the output.

https://facebook.github.io/relay/graphql/mutations.htm has a decent writeup